spark/v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/MergeRowsExec.scala

```diff
@@ -148,24 +138,27 @@ case class MergeRowsExec(
       }
     }
 
-    var lastMatchedRowId: InternalRow = null
+    val rowIdAttrOrdinal = if (performCardinalityCheck) {
```

Comment:
Can we move this into
`def processRowWithCadinalityCheck`?

Reply:
Actually maybe not ... Mainly i just want to remove the branch if we can. Maybe just make this an option?

Reply:
Unlike `processRowWithCadinalityCheck`, this is not called per every row. It is called once per task so having a branch should not impact the performance.